chef/cookbooks/nova/recipes/ceph.rb

```diff
@@ -0,0 +1,185 @@
+#
+# Copyright (c) 2015 SUSE Linux GmbH.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Cookbook Name:: nova
+# Recipe:: ceph
+#
+
+has_internal = false
+has_external = false
+
+cinder_controller = search(:node, "roles:cinder-controller")
```

Comment:
Actually, you can even make it `search_env_filtered(:node, "roles:cinder-controller").first` and change the test on the next line, and replace `cinder_controller[0]` with `cinder_controller`